R/auxiliary.R

```diff
@@ -184,3 +184,243 @@ get_date_from_test_periods <- function(GeoLift, treatment_end_date, frequency =
+#' geographic unit. It requires a "locations" column with the geo name,
+#' a "Y" column with the outcome data (units), a time column with the indicator
+#' of the time period (starting at 1), and covariates.
+#' @param locs list of test locations. If none are provided, only the aggregated
+#' KPI (total) will be calculated and appended. Set to NULL by default.
+#'
+#' @return
+#' A data frame with additional values for the test regions (combined_test)
+#' and total KPI (total).
+#'
+#' @export
+AppendAgg <- function(data, locs = NULL){
+
+  if(is.null(locs)){
+    aux <- data %>%
+      dplyr::group_by(time) %>%
+      dplyr::summarise(Y = sum(Y))
+    aux$location <- "control_markets"
+    aux <- dplyr::bind_rows(data, aux)
+  } else if(!(all(tolower(locs) %in% tolower(unique(data$location))))){
+    message("Please specify a valid vector of location names.")
+    return(NULL)
+  } else{
+    aux <- data %>%
+      dplyr::filter(!(location %in% locs)) %>%
+      dplyr::group_by(time) %>%
+      dplyr::summarise(Y = sum(Y))
+    aux$location <- "control_markets"
+  }
+
+
+  if(all(tolower(locs) %in% tolower(unique(data$location)))){
+    auxCombo <- data %>%
+      dplyr::filter(location %in% locs) %>%
+      dplyr::group_by(time) %>%
+      dplyr::summarise(Y = sum(Y))
+    auxCombo$location <- "test_markets"
+    aux <- dplyr::bind_rows(aux, auxCombo)
+  } else{
+    message("Please specify a valid vector of location names.")
+    return(NULL)
+  }
+
+  return(as.data.frame(aux))
+
+}
+
+#' Auxiliary function to calculate correlations between input markets.
+#'
+#' @description
+#'
+#' Auxiliary function to calculate correlations between input markets.
+#'
+#' @param data A data.frame containing the historical conversions by
+#' geographic unit. It requires a "locations" column with the geo name,
+#' a "Y" column with the outcome data (units), a time column with the indicator
+#' of the time period (starting at 1), and covariates.
+#' @param location_id Name of the location variable (String).
+#' @param time_id Name of the time variable (String).
+#' @param Y_id Name of the outcome variable (String).
+#' @param dtw Emphasis on Dynamic Time Warping (DTW), dtw = 1 focuses exclusively
+#' on this metric while dtw = 0 (default) relies on correlations only.
+#' @param markets List of markets to use in the calculation of the correlations.
+#'
+#' @return
+#' MarketMatching object.
+#'
+#' @export
+MarketCorrelations <- function(data,
+                               location_id = "location",
+                               time_id = "time",
+                               Y_id = "Y",
+                               dtw = 0,
+                               markets = NULL) {
+  data <- data %>% dplyr::rename(Y = paste(Y_id), location = paste(location_id), time = paste(time_id))
+  data$location <- tolower(data$location)
+  astime <- seq(as.Date("2000/1/1"), by = "day", length.out = max(data$time))
+  data$astime <- astime[data$time]
```

Comment:
Since we haven't really defined a `GeoDataRead()` object nor have we restricted our functions to only run with these sort of objects it is possible that some people might provide a data set that didn't went through that function. Right now the only requirement is that the data input is a data frame.